cheroot/ssl/pyopenssl.py

```diff
@@ -150,14 +150,24 @@ def readline(self, size=-1):
             size,
         )
 
-    def sendall(self, *args, **kwargs):
-        """Send whole message to the socket."""
+    def read(self, *args, **kwargs):
+        """Read from the wrapped socket, with retry."""
         return self._safe_call(
-            False,
-            super(SSLFileobjectMixin, self).sendall,
+            True,
+            super(SSLFileobjectMixin, self).read,
             *args, **kwargs
         )
 
+    def sendall(self, *args, **kwargs):
+        """Send whole message to the socket. Unsupported, do not use."""
+        # Not supported due to https://github.com/pyca/pyopenssl/issues/176.
+        # Until that bug is fixed, sendall() may throw SSL.WantWriteError, but
+        # there is no correct way to retry the call because we don't know how
+        # many bytes were already transmitted. We could work around this by
+        # reimplementing sendall() using send(), but we don't actually use
+        # sendall() anywhere.
+        raise NotImplementedError('sendall() is unsupported by pyOpenSSL')
```

Comment:
I've given this some thought. I think we shouldn't proactively block this but rather call, catch the errors and turn those errors into `RuntimeError` referring to the explanation that it will work sometimes and sometimes it won't.

Reply:
In general I would much prefer a randomly failing function to just never work than randomly work, but the only thing that really matters to me at the moment is fixing #245, so whatever will cause the PR to be accepted.

Reply:
I think this has to be taken out into a separate PR. This is one of the problems of non-atomic PRs: many things can block them which they could've been debated separately.

Reply:
fine, undone